core/codegen/src/bang/websocket.rs

```diff
@@ -0,0 +1,56 @@
+use devise::{Level, Diagnostic, Spanned};
+use proc_macro2::TokenStream;
+use syn::{parse::{Parse, ParseStream, discouraged::Speculative}, Pat};
+
+pub fn _macro(input: proc_macro::TokenStream) -> devise::Result<TokenStream> {
+    let closure: syn::ExprClosure = syn::parse(input)?;
+
+    if closure.inputs.len() != 1 {
+        return Err(Diagnostic::spanned(
+            closure.inputs.span(),
+            Level::Error,
+            "rocket::response::websocket::CreateWebsocket! needs exactly one closure input"
+        ));
+    }
+
+    if closure.capture.is_none() {
+        return Err(
+            Diagnostic::spanned(
+                closure.span(),
+                Level::Error,
+                "rocket::response::websocket::CreateWebsocket! needs an closure that captures it's inputs"
+            )
+                .span_help(closure.or1_token.span(), "add the `move` keyword to the closure")
+        );
+    }
```

Comment:
Without this check, does the error provided by the compiler is clear enough ?
If this is the case, I think we should let the compiler do it's job. Otherwise, keep this.

Reply:
While the core diagnostic about a async block that outlives a borrowed value is correct, the suggestion to fix this isn't:
```
error[E0373]: async block may outlive the current function, but it borrows `channel`, which is owned by the current function
  --> src/main.rs:20:5
   |
20 | /     CreateWebsocket! {
21 | |         |mut channel| {
22 | |             while let Some(msg) = channel.next().await {
   | |                                   ------- `channel` is borrowed here
23 | |                 println!("handler {msg:?}")
24 | |             }
25 | |         }
26 | |     }
   | |_____^ may outlive borrowed value `channel`
   |
note: async block is returned here
  --> src/main.rs:20:5
   |
20 | /     CreateWebsocket! {
21 | |         |mut channel| {
22 | |             while let Some(msg) = channel.next().await {
23 | |                 println!("handler {msg:?}")
24 | |             }
25 | |         }
26 | |     }
   | |_____^
   = note: this error originates in the macro `CreateWebsocket` (in Nightly builds, run with -Z macro-backtrace for more info)
help: to force the async block to take ownership of `channel` (and any other referenced variables), use the `move` keyword
   |
20 |     move CreateWebsocket! {
   |     ++++

For more information about this error, try `rustc --explain E0373`.
```
So to help users to better understand whats failing and how to fix it, I would keep the custom error.